test/extended/builds/run_policy.go

```diff
@@ -0,0 +1,241 @@
+package builds
+
+import (
+	"fmt"
+	"strings"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	buildclient "github.com/openshift/origin/pkg/build/client"
+	buildutil "github.com/openshift/origin/pkg/build/util"
+	exutil "github.com/openshift/origin/test/extended/util"
+	kapi "k8s.io/kubernetes/pkg/api"
+)
+
+var _ = g.Describe("[builds][Slow] using build configuration runPolicy", func() {
+	defer g.GinkgoRecover()
+	var (
+		// Use invalid source here as we don't care about the result
+		oc = exutil.NewCLI("cli-build-run-policy", exutil.KubeConfigPath())
+	)
+
+	g.JustBeforeEach(func() {
+		g.By("waiting for builder service account")
+		err := exutil.WaitForBuilderAccount(oc.KubeREST().ServiceAccounts(oc.Namespace()))
+		o.Expect(err).NotTo(o.HaveOccurred())
+		// Create all fixtures
+		oc.Run("create").Args("-f", exutil.FixturePath("..", "extended", "fixtures", "run_policy")).Execute()
+	})
+
+	g.Describe("build configuration with Parallel build run policy", func() {
+		g.It("runs the builds in parallel", func() {
+			g.By("starting multiple builds")
+			var (
+				startedBuilds []string
+				counter       int
+			)
+			bcName := "sample-parallel-build"
+
+			buildWatch, err := oc.REST().Builds(oc.Namespace()).Watch(kapi.ListOptions{
+				LabelSelector: buildutil.BuildConfigSelector(bcName),
+			})
+			defer buildWatch.Stop()
+
+			// Start first build
+			out, err := oc.Run("start-build").Args(bcName).Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(strings.TrimSpace(out)).ShouldNot(o.HaveLen(0))
+			startedBuilds = append(startedBuilds, strings.TrimSpace(out))
+
+			// Wait for it to become running
+			for {
+				event := <-buildWatch.ResultChan()
+				build := event.Object.(*buildapi.Build)
+				o.Expect(buildutil.IsBuildComplete(build)).Should(o.BeFalse())
+				if build.Name == startedBuilds[0] && build.Status.Phase == buildapi.BuildPhaseRunning {
+					break
+				}
+			}
+
+			for i := 0; i < 2; i++ {
+				out, err := oc.Run("start-build").Args(bcName).Output()
+				o.Expect(err).NotTo(o.HaveOccurred())
+				o.Expect(strings.TrimSpace(out)).ShouldNot(o.HaveLen(0))
+				startedBuilds = append(startedBuilds, strings.TrimSpace(out))
+			}
+
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			for {
+				event := <-buildWatch.ResultChan()
+				build := event.Object.(*buildapi.Build)
+				if build.Name == startedBuilds[0] {
+					if buildutil.IsBuildComplete(build) {
+						break
+					}
```

Comment:
you need at least one pre-check to ensure you went through the loop at least one time while build1 was pending.  Otherwise if build1 completes extremely fast, you'll never check the other 2 builds to ensure they were in pending.

something like:  if loopcount==0  && IsBuildComplete(build) -> fail test  (because now you have no way to confirm whether or not the other 2 builds would have been properly queued).